.github/workflows/cuda.yml

```diff
@@ -14,13 +14,9 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-# Ref.:
-#   https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/ubuntu18.04/10.1/base/Dockerfile
-#   https://github.com/ComputationalRadiationPhysics/picongpu/blob/0.5.0/share/picongpu/dockerfiles/ubuntu-1604/Dockerfile
-#   https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/
   build_nvcc:
-    name: NVCC 11.3 SP
-    runs-on: ubuntu-20.04
+    name: NVCC SP
+    runs-on: ubuntu-24.04
```

Comment:
I would not go over 22.04, because that way we do not cover older CUDA compilers anymore and will likely loose compatibility.

Instead, please do something like this https://github.com/BLAST-ImpactX/impactx/pull/869 and update our documentation that we only support CUDA 11.7+ now.

Reply:
Okay, done in [dd2fe9b](https://github.com/ECP-WarpX/WarpX/pull/5731/commits/dd2fe9bbf30489649794aaa771b414b98a6380a7).

The documentation does not seem to need any update, it already states 11.7+, if I'm not mistaken.

I initially chose to go for 12.6+ (and Ubuntu 24.04) because I saw that this is what they did in https://github.com/AMReX-Codes/amrex/pull/4330 in AMReX.

In general isn't it a risk to mix Ubuntu 22.04 and Ubuntu 24.04 support in our CI coverage, i.e., don't we risk to have potentially holes in how we test compatibility? Shoudn't we then simply cover only Ubuntu 22.04 until it reaches EOL? I'm just trying to understand which of the two strategies, the mixed one we are setting up here or the single-version one they set up in AMReX, makes more sense, if any.

Reply:
> The documentation does not seem to need any update, it already states 11.7+, if I'm not mistaken.

![Screenshot from 2025-03-06 22-18-46](https://github.com/user-attachments/assets/e0164acc-ec3a-465b-bcf5-90231fb3b55c)